pkg/cli/httpClient.go

```diff
@@ -0,0 +1,183 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	zotErrors "github.com/anuvu/zot/errors"
+)
+
+var httpClient *http.Client = createHTTPClient() //nolint: gochecknoglobals
+
+const httpTimeout = 5 * time.Second
```

Comment:
Ok, so the http client has a 5 sec timeout. Did you verify the error (timeout) path?

Reply:
Yes. It fails and returns with an error on timeout